scheduler/src/test/java/org/cloudfoundry/autoscaler/scheduler/quartz/AppScalingScheduleJobTest.java

```diff
@@ -0,0 +1,492 @@
+
+		JobDataMap jobDataMap = setupJobData(jobInformation.getJobDetail(), JobActionEnum.START);
+		ActiveScheduleEntity activeScheduleEntity = ScheduleJobHelper.setupActiveSchedule(jobDataMap);
+
+		HttpURLConnection mockHttpURLConnection = Mockito.mock(HttpURLConnection.class);
+		Mockito.when(scalingEngineUtil.getConnection(Mockito.anyObject(), Mockito.anyObject()))
+				.thenReturn(mockHttpURLConnection);
+		Mockito.when(mockHttpURLConnection.getResponseCode()).thenReturn(500);
+		Mockito.when(mockHttpURLConnection.getResponseMessage()).thenReturn("test error message");
+
+		scheduler.scheduleJob(jobInformation.getJobDetail(), jobInformation.getTrigger());
+
+		//sleep
+		Thread.sleep(TimeUnit.SECONDS.toMillis(2));
+
+		Mockito.verify(activeScheduleDao, Mockito.times(1)).create(activeScheduleEntity);
+		Mockito.verify(mockAppender, Mockito.atLeastOnce()).append(logCaptor.capture());
+		String expectedMessage = messageBundleResourceHelper.lookupMessage("scalingengine.notification.failed", 500,
+				"test error message", activeScheduleEntity.getAppId(), activeScheduleEntity.getId(),
+				JobActionEnum.START);
+
+		assertThat("Log level should be ERROR", logCaptor.getValue().getLevel(), is(Level.ERROR));
+		assertThat(logCaptor.getValue().getMessage().getFormattedMessage(), is(expectedMessage));
+
+	}
+
+	@Test
+	public void testNotifyEndOfActiveScheduleToScalingEngine_when_responseError() throws Exception {
+		setLogLevel(Level.ERROR);
+		// Build the job and trigger
+		JobInformation jobInformation = new JobInformation<>(AppScalingScheduleEndJob.class);
+
+		JobDataMap jobDataMap = setupJobData(jobInformation.getJobDetail(), JobActionEnum.END);
+		ActiveScheduleEntity activeScheduleEntity = ScheduleJobHelper.setupActiveSchedule(jobDataMap);
+
+		HttpURLConnection mockHttpURLConnection = Mockito.mock(HttpURLConnection.class);
+		Mockito.when(scalingEngineUtil.getConnection(Mockito.anyObject(), Mockito.anyObject()))
+				.thenReturn(mockHttpURLConnection);
+		Mockito.when(mockHttpURLConnection.getResponseCode()).thenReturn(500);
+		Mockito.when(mockHttpURLConnection.getResponseMessage()).thenReturn("test error message");
+
+		scheduler.scheduleJob(jobInformation.getJobDetail(), jobInformation.getTrigger());
+
+		//sleep
+		Thread.sleep(TimeUnit.SECONDS.toMillis(2));
+
+		Mockito.verify(activeScheduleDao, Mockito.times(1)).delete(activeScheduleEntity.getId());
+		Mockito.verify(mockAppender, Mockito.atLeastOnce()).append(logCaptor.capture());
+		String expectedMessage = messageBundleResourceHelper.lookupMessage("scalingengine.notification.failed", 500,
+				"test error message", activeScheduleEntity.getAppId(), activeScheduleEntity.getId(), JobActionEnum.END);
+
+		assertThat("Log level should be ERROR", logCaptor.getValue().getLevel(), is(Level.ERROR));
+		assertThat(logCaptor.getValue().getMessage().getFormattedMessage(), is(expectedMessage));
+
+	}
+
+	@Test
+	public void testNotifyScalingEngine_when_invalidURL() throws Exception {
```

Comment:
An invalid URL would fail differently.
There is the case where I give you a garbage url, e.g., ftp://wrong.scheme
Or an non-existent endpoint, e.g., http://going.no.where

None of these return a 404.